Tests/WeaverTests.cs

```diff
@@ -156,7 +170,18 @@ public void UniquePropertyTypeButExplicitMethodName_WithIsEmittedWithLongName()
     public void OriginalWithMethodIsRemoved()
     {
         var type1 = assembly.GetType("AssemblyToProcess.PrimitiveValues");
-        Assert.False(type1.GetMethods().Any(m => m.Name == "With" && m.IsGenericMethod));
+        // check for the generic With method
+        Assert.False(type1.GetMethods().Any(m =>
+            m.IsPublic &&
+            m.Name == "With" &&
+            m.IsGenericMethod &&
+            m.GetParameters().Length == 1));
+        // check for a With method with Object parameter
+        Assert.False(type1.GetMethods().Any(m => 
+            m.IsPublic &&
+            m.Name == "With" && 
+            m.GetParameters().Length == 1 &&
+            m.GetParameters()[0].ParameterType == typeof(object)));
```

Comment:
I believe this change is not related to multiple constructor support. So,
a) Remove it from this PR and add a new PR if you need it.
b) Provide some motivation of why both generic and object need to be supported.
c) Add test cases.